scrunch/datasets.py

```diff
@@ -262,28 +295,96 @@ def __str__(self):
         return self.email
 
 
+class Members:
+    """
+    Class for handling collections of members. Adding users,
+    removing users, listing users on a resource...
+    """
+
+    def __init__(self, resource):
+        self.resource = resource
+
+    def list(self):
+        """
+        :return: A list of members of the Entity
+        """
+        members = []
+        for member in self.resource.members.index.values():
+            # members can be users or teams
+            user = member.get('email')
+            if not user:
+                user = member.get('name')
+            members.append(user)
+        return members
+
+    def _validate_member(self, member):
+        """
+        Validate and instanciate if necessary a member as 
+        Team or User
+        """
+        if isinstance(member, User) or isinstance(member, Team):
+            return member    
+        try:
+            member = get_user(member)
+        except KeyError:
+            try:
+                member = get_team(member)
+            except:
+                raise KeyError('Member %s is not a Team nor a User' % member)
+        return member
+
+    def remove(self, member):
+        """
+        :param member: email, User instance, team name or Team instance
+        :return: None
+        """
+        member = self._validate_member(member)
+        self.resource.members.patch({member.url: None})
+
+    def add(self, member, edit=False):
+        """
+        :param member: email, User instance, team name or Team instance
+        :return: None
+        """
+        member = self._validate_member(member)
```

Comment:
Note that the API allows to add members directly by email and if they don't exist they will be created as new user accounts on Crunch. Doing `_validate_member` here removes that option to "invite" new members by email directly since this would raise for a non existing new user.

Reply:
Look how this method is designed, since the beginnings of time. We are patching the `/members/` resource with `self.resource.members.patch({member.url: {'edit': edit}})` So it needs to be an instance with a url. I'm happy to explore other options, but I believe this is to be addressed on a different independent ticket.

Reply:
Great then, if it's been like that we could add another method later to invite.